tests/unit/restapi/lib/actions.py

```diff
@@ -0,0 +1,88 @@
+# This Software (Dioptra) is being made available as a public service by the
+# National Institute of Standards and Technology (NIST), an Agency of the United
+# States Department of Commerce. This software was developed in part by employees of
+# NIST and in part by NIST contractors. Copyright in portions of this software that
+# were developed by NIST contractors has been licensed or assigned to NIST. Pursuant
+# to Title 17 United States Code Section 105, works of NIST employees are not
+# subject to copyright protection in the United States. However, NIST may hold
+# international copyright in software created by its employees and domestic
+# copyright (or licensing rights) in portions of software that were assigned or
+# licensed to NIST. To the extent that NIST holds copyright in this software, it is
+# being made available under the Creative Commons Attribution 4.0 International
+# license (CC BY 4.0). The disclaimers of the CC BY 4.0 license apply to all parts
+# of the software developed or licensed by NIST.
+#
+# ACCESS THE FULL CC BY 4.0 LICENSE HERE:
+# https://creativecommons.org/licenses/by/4.0/legalcode
+"""Shared actions for REST API unit tests.
+
+This module contains shared actions used across test suites for each of the REST
+API endpoints.
+"""
+
+from typing import Any
+
+from flask.testing import FlaskClient
+from werkzeug.test import TestResponse
+
+from dioptra.restapi.routes import (
+    V1_AUTH_ROUTE,
+    V1_QUEUES_ROUTE,
+    V1_ROOT,
+    V1_USERS_ROUTE,
+)
+
+
+def login(client: FlaskClient, username: str, password: str) -> TestResponse:
+    return client.post(
+        f"/{V1_ROOT}/{V1_AUTH_ROUTE}/login",
+        json={"username": username, "password": password},
+    )
+
+
+def register_user(client: FlaskClient, username: str, email: str) -> dict[str, Any]:
+    password = "supersecurepassword"
+    response = client.post(
+        f"/{V1_ROOT}/{V1_USERS_ROUTE}",
+        json={
+            "username": username,
+            "email": email,
+            "password": password,
+            "confirmPassword": password,
+        },
+        follow_redirects=True,
+    ).get_json()
+    return {
+        "user_id": response["id"],
+        "username": response["username"],
+        "password": password,
+        "email_address": response["email"],
+        "default_group_id": response["groups"][0]["id"],
+    }
+
+
+def register_queue(
+    client: FlaskClient,
+    name: str,
+    group_id: int,
+    description: str = "",
```

Comment:
Minor nitpick: The description field should be optional in the API since it will be nullable in the database. So if you do this,

```suggestion
    description: str | None = None,
```

Then you can use an if statement to optionally add the description field to the payload.

The rationale for this is so we have one way to signify that a description is empty/blank, otherwise we'll need to do compound checks, i.e. "a description is considered empty if the description field is null or its a zero-length string or its a string that only contains whitespace."